openrtb_ext/seat_non_bids_test.go

```diff
@@ -0,0 +1,240 @@
+			fields: fields{seatNonBidsMap: nil},
+			args:   args{nonBidparams: NonBidParams{Bid: &openrtb2.Bid{ImpID: "imp"}, Seat: "bidder1"}},
+			want: map[string][]NonBid{
+				"bidder1": {{ImpId: "imp"}},
+			},
+		},
+		{
+			name: "multiple-nonbids-for-same-seat",
+			fields: fields{
+				seatNonBidsMap: map[string][]NonBid{
+					"bidder2": {{ImpId: "imp1"}},
+				},
+			},
+			args: args{nonBidparams: NonBidParams{Bid: &openrtb2.Bid{ImpID: "imp2"}, Seat: "bidder2"}},
+			want: map[string][]NonBid{
+				"bidder2": {{ImpId: "imp1"}, {ImpId: "imp2"}},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			snb := &NonBidsWrapper{
+				seatNonBidsMap: tt.fields.seatNonBidsMap,
+			}
+			snb.AddBid(tt.args.nonBidparams)
+			assert.Equalf(t, tt.want, snb.seatNonBidsMap, "expected seatNonBidsMap not nil")
+		})
+	}
+}
+
+func TestSeatNonBidsGet(t *testing.T) {
+	type fields struct {
+		snb *NonBidsWrapper
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		want   []SeatNonBid
+	}{
+		{
+			name:   "nil seat nonbids",
+			fields: fields{nil},
+		},
+		{
+			name: "get seat nonbids",
+			fields: fields{&NonBidsWrapper{
+				seatNonBidsMap: map[string][]NonBid{
+					"bidder1": {
+						{
+							ImpId:      "imp1",
+							StatusCode: 100,
+						},
+					},
+					"bidder2": {
+						{
+							ImpId:      "imp2",
+							StatusCode: 200,
+						},
+					},
+				},
+			}},
+			want: []SeatNonBid{
+				{
+					Seat: "bidder1",
+					NonBid: []NonBid{
+						{
+							ImpId:      "imp1",
+							StatusCode: 100,
+						},
+					},
+				},
+				{
+					Seat: "bidder2",
+					NonBid: []NonBid{
+						{
+							ImpId:      "imp2",
+							StatusCode: 200,
+						},
+					},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := tt.fields.snb.Get(); !assert.ElementsMatch(t, tt.want, got) {
+				t.Errorf("seatNonBids.Get() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestSeatNonBidsMerge(t *testing.T) {
+	type target struct {
+		snb *NonBidsWrapper
+	}
+	tests := []struct {
+		name   string
+		fields target
+		input  NonBidsWrapper
+		want   *NonBidsWrapper
+	}{
+		{
+			name:   "target NonBidsWrapper is nil",
+			fields: target{nil},
+			want:   nil,
+		},
+		{
+			name:   "input NonBidsWrapper contains nil map",
+			fields: target{&NonBidsWrapper{}},
+			input:  NonBidsWrapper{seatNonBidsMap: nil},
+			want:   &NonBidsWrapper{},
+		},
+		{
+			name:   "input NonBidsWrapper contains empty nonBids",
+			fields: target{&NonBidsWrapper{}},
+			input:  NonBidsWrapper{seatNonBidsMap: make(map[string][]NonBid)},
+			want:   &NonBidsWrapper{},
+		},
+		{
+			name:   "merge nonbids in empty target NonBidsWrapper",
+			fields: target{&NonBidsWrapper{}},
+			input: NonBidsWrapper{
+				seatNonBidsMap: map[string][]NonBid{
+					"pubmatic": {
+						{
+							ImpId:      "imp1",
+							StatusCode: 100,
+						},
+					},
+				},
+			},
```

Comment:
@ashishshinde-pubm : Use `sampleSeatNonBidMap` helper instead for `seatNonBidsMap` construction. It will make the test readable

Reply:
done